File: backend/TalentSyncAI.Backend/TalentSyncAI.Api/Models/Entities/Interview.cs
using TalentSyncAI.Api.Models.Enums;

namespace TalentSyncAI.Api.Models.Entities
{
    public class Interview
    {
        public int Id { get; set; }

        public int JobApplicationId { get; set; }

        public string Title { get; set; } = string.Empty;

        public DateTime ScheduledAt { get; set; }

        public int DurationMinutes { get; set; } = 60;

        public string MeetingLink { get; set; } = string.Empty;

        public string Location { get; set; } = string.Empty;

        public string Notes { get; set; } = string.Empty;

        public InterviewStatus Status { get; set; }
            = InterviewStatus.Scheduled;

        public DateTime CreatedAt { get; set; }
            = DateTime.UtcNow;

        public DateTime UpdatedAt { get; set; }
            = DateTime.UtcNow;

        public JobApplication JobApplication { get; set; }
            = null!;
    }
}

File: frontend/src/types/interview.ts
export type InterviewStatus =
  | "Scheduled"
  | "Rescheduled"
  | "Completed"
  | "Cancelled"
  | "NoShow";

export interface CreateInterviewRequest {
  jobApplicationId: number;
  title: string;
  scheduledAt: string;
  durationMinutes?: number;
  meetingLink?: string;
  location?: string;
  notes?: string;
}

export interface UpdateInterviewRequest {
  title: string;
  scheduledAt: string;
  durationMinutes: number;
  meetingLink: string;
  location: string;
  notes: string;
  status: InterviewStatus;
}

export interface InterviewResponse {
  id: number;
  jobApplicationId: number;
  jobId: number;
  jobTitle: string;
  candidateUserId: number;
  candidateName: string;
  candidateEmail: string;
  title: string;
  scheduledAt: string;
  durationMinutes: number;
  meetingLink: string;
  location: string;
  notes: string;
  status: InterviewStatus;
  createdAt: string;
  updatedAt: string;
}
